### ISSUE TITLE
Heroku deploy fails - on verify, server not reachable?

### ISSUE THREAD

**Issue by thebucknerlife** [ForestAdmin/lumber#165]
Hi,

Ever time I try to follow these direction (https://doc.forestadmin.com/developer-guide/lumber.html#deploy-to-production) to deploy to Heroku I get the same error on clicking "verify":

```
foo-admin
Your application is not reachable. Please, double-check that you installed Forest properly, your application URL is correct and your server is running.
```

I setup a fresh Heroku app on my account, added `FOREST_AUTH_SECRET` and `FOREST_ENV_SECRET` to config and then pasted the full heroku application https URL into the admin.

What gives? Maybe something missing in the docs or some other issue I need to address? When I visit my application URL I get the standard Heroku new app page "Heroku | Welcome to your new app!".

How should I proceed? Thanks.

Lumber looks great. Thank you for making this - I'm excited to dig into it more.

Unblocking this would be extremely helpful and timely in showing the rest of our team the admin. We're making some decisions today about how to implement out admin.

**Comment by arnaudbesnier** [comment]
Hi @thebucknerlife , on our side, it looks like your production environment is not setup at all.

What code is running on this new Heroku app?
Did you deploy your Admin server (the node server generated by Lumber) on your fresh Heroku app?

Thanks!

**Comment by thebucknerlife** [comment]
hey @arnaudbesnier I don't quite understand. No I didn't deploy it.

Thinking further, I see the issue. Your directions don't clear state "once you've committed and deployed your lumber code to Heroku, then...". I'd suggest you make that a lot more clear here: https://doc.forestadmin.com/developer-guide/lumber.html#deploy-to-production

In any case, I pushed my code to my Heroku app. I still have the same error, however.

When I pasted my heroku app into the config and press "Verify", I get `Your application is not reachable. Please, double-check that you installed Forest properly, your application URL is correct and your server is running.`

What is the next step to take? Thanks

**Comment by frostini** [comment]
@thebucknerlife just had the same issue, hope this helps !

the application autogenerated a local .env, this sets the server port via the ENV variable named APPLICATION_PORT.  This is a problem because heroku dynamically sets a port via the process.env.PORT variable/port our server should be listening on. It's listed as the #1 issue on #R10 errors [https://help.heroku.com/P1AVPANS/why-is-my-node-js-app-crashing-with-an-r10-error](article). Took me a bit to understand what it meant.

To get it working you can:

- rename APPLICATION_PORT to PORT in both your .env file and the bin/www file
- remove any heroku CONFIG variables related to APPLICATION_PORT or PORT
- commit and re-deploy your changes
- check heroku logs that ```2019-08-01T23:31:19.684729+00:00 heroku[web.1]: State changed from starting to up```
- and you hopefully you should be :ok_hand:
*the port ended up being dynamically set at ```2019-08-01T23:31:19.138074+00:00 app[web.1]: Your application is listening on port 7815.```

**Comment by frostini** [comment]
@arnaudbesnier renaming the APPLICATION_PORT .env var to PORT should suffice in closing this issue out :+1:

**Comment by arnaudbesnier** [comment]
This issue should be fixed with the latest Lumber version (v2.3.11).

Here is the fix:
https://github.com/ForestAdmin/lumber/commit/bc68baf4cfb37c874abc246e430b9e3fd9ee02ae

Thanks for you feedback 🌲
